### PR TITLE
[Issue-23363] buf2hexstr_sep ebcdic2ascii translation off by 1

### PR DESCRIPTION
**Pull request by shahsb**
**Issue:**

buf2hexstr_sep() function was updated in OpenSSL 3.0 so that the buffer q points to the last character in the buffer buf. Because of this change, the calculation for ebcdic2ascii is off by 1. q - str - 1 is 1 less than the length of buf, which results in all characters but the last to be translated in buf. 

**Solution:**
changing to ebcdic2ascii(str, str, q - str);

#23363

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Next time, please use `Fixes #23363` on a separate line in the commit message body instead of mentioning it in the subject line.

**Comment by shahsb** [comment]
> Next time, please use `Fixes #23363` on a separate line in the commit message body instead of mentioning it in the subject line.

Sure, Thanks!

**Comment by shahsb** [comment]
This pull request is ready for merging. @t8m, could you please proceed with the merge?

**Comment by t8m** [comment]
@shahsb no, not really. It needs a second approval. PRs are merged when they have `approval: ready to merge` label.

**Comment by shahsb** [comment]
@t8m - This PR is pending from long time. How to request approval from core team?

**Review by abaez004 — APPROVED** [review]

**Comment by tom-cosgrove-arm** [comment]
I'm closing this as it's a duplicate of #23404. Apologies for not noticing sooner

**Pull request closed without merging**
